plugins/inputs/procstat/native_finder.go

```diff
@@ -80,6 +80,41 @@ func (pg *NativeFinder) FullPattern(pattern string) ([]PID, error) {
 	return pids, err
 }
 
+// ChildPattern matches children pids on the command line when the process was executed
+func (pg *NativeFinder) ChildPattern(pattern string) ([]PID, error) {
+	var parentpids []PID
+	var pids []PID
+	regxPattern, err := regexp.Compile(pattern)
+	if err != nil {
+		return pids, err
+	}
+	procs, err := process.Processes()
+	if err != nil {
+		return pids, err
+	}
+	for _, p := range procs {
+		cmd, err := p.Cmdline()
+		if err != nil {
+			continue
+		}
+		if regxPattern.MatchString(cmd) {
+			parentpids = append(parentpids, PID(p.Pid))
+		}
+	}
+	for _, p := range parentpids {
+		parent, err := process.NewProcess(int32(p))
+		if err != nil {
+			return pids, err
+		}
+		children, _ := parent.Children()
+
+		for _, child := range children {
+			pids = append(pids, PID(child.Pid))
+		}
+	}
+	return pids, err
```

Comment:
> How about

It looks good! But there's a small problem?
The following code cannot pass the unit test because it always returns error 'unable to get children of process *** : exit status 1'
```
children, err := parent.Children()
		 if err != nil {
			fmt.Println("unable to get children of process", p.Pid, ":", err)
		}
```
```suggestion
var pids []PID
	regxPattern, err := regexp.Compile(pattern)
	if err != nil {
		return pids, err
	}
	procs, err := process.Processes()
	if err != nil {
		return pids, err
	}

	for _, p := range procs {
		cmd, err := p.Cmdline()
		if err != nil || !regxPattern.MatchString(cmd) {
			continue
		}
		parent, err := process.NewProcess(p.Pid)
		if err != nil {
			return nil, fmt.Errorf("unable to get process %d: %w", p.Pid, err)
		}
		children, err := parent.Children()
		if err != nil {
			fmt.Println("unable to get children of process", p.Pid, ":", err)
		}
		for _, child := range children {
			pids = append(pids, PID(child.Pid))
		}
	}
	return pids, err
```

Reply:
Which test is it? Both code pieces are logically equivalent, do you agree? So if I didn't miss anything this is a race condition in the test as the new code now directly looks for the children while the other one did is a _bit_ later...